server/server/src/lib.rs

```diff
@@ -91,6 +108,45 @@ macro_rules! internal {
     }};
 }
 
+impl From<SimplifiedStripeError> for ServerError<SwitchAccountTierError> {
+    fn from(e: SimplifiedStripeError) -> Self {
+        match e {
+            SimplifiedStripeError::CardDeclined(decline_type) => {
+                ClientError(SwitchAccountTierError::CardDeclined(decline_type))
+            }
+            SimplifiedStripeError::InvalidCreditCard(field) => {
+                ClientError(SwitchAccountTierError::InvalidCreditCard(field))
+            }
+            SimplifiedStripeError::Other(msg) => internal!("{}", msg),
+        }
+    }
+}
+
+impl From<stripe::WebhookError> for ServerError<StripeWebhookError> {
+    fn from(e: WebhookError) -> Self {
+        match e {
+            WebhookError::BadKey => {
+                internal!("Cannot verify stripe request because server is using a bad signing key.")
```

Comment:
Can someone external force this error?

Reply:
No, someone can't. This error only happens if the stripe secret signing key we have (which is a part of our environment variables) is invalid.